src/AWS/Shared/Storage/DynamoDBStorage.cs

```diff
@@ -163,34 +165,143 @@ private async Task CreateTable(string tableName, List<KeySchemaElement> keys, Li
 
             try
             {
-                var response = await ddbClient.CreateTableAsync(request);
-                TableDescription description = null;
-                do
+                await ddbClient.CreateTableAsync(request);
+                TableDescription description = await TableWaitOnStatusAsync(tableName, TableStatus.CREATING, TableStatus.ACTIVE);
+                await TableUpdateTtlAsync(tableName, ttlAttributeName);
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(ErrorCode.StorageProviderBase, $"Could not create table {tableName}", exc);
+                throw;
+            }
+        }
+
+        private async Task TableUpdateTtlAsync(string tableName, string ttlAttributeName)
+        {
+            if (!string.IsNullOrEmpty(ttlAttributeName))
+            {
+                return;
+            }
+
+            await ddbClient.UpdateTimeToLiveAsync(new UpdateTimeToLiveRequest
+            {
+                TableName = tableName,
+                TimeToLiveSpecification = new TimeToLiveSpecification { AttributeName = ttlAttributeName, Enabled = true }
+            });
+
+            await TableWaitOnStatusAsync(tableName, TableStatus.UPDATING, TableStatus.ACTIVE);
+        }
+
+        private async Task UpdateTableAsync(TableDescription tableDescription, List<AttributeDefinition> attributes, List<GlobalSecondaryIndex> secondaryIndexes = null, string ttlAttributeName = null)
+        {
+            var request = new UpdateTableRequest
```

Comment:
Nit: The request object doesn't need to be created if nothing changes. Does make the check for changes a bit more complicated though.

Reply:
That is exactly why I left it in place. I'm compiling the desired state using common logic and then just comparing the desired state with the actual state.

I'd prefer to leave this as is if you don't have a strong opinion.